atom/browser/atom_blob_reader.cc

```diff
@@ -53,22 +53,21 @@ AtomBlobReader::AtomBlobReader(content::ChromeBlobStorageContext* blob_context)
 
 AtomBlobReader::~AtomBlobReader() {}
 
-void AtomBlobReader::StartReading(
-    const std::string& uuid,
-    const AtomBlobReader::CompletionCallback& completion_callback) {
+void AtomBlobReader::StartReading(const std::string& uuid,
+                                  atom::util::Promise promise) {
   DCHECK_CURRENTLY_ON(BrowserThread::IO);
 
   auto blob_data_handle = blob_context_->context()->GetBlobDataFromUUID(uuid);
-  auto callback = base::Bind(&RunCallbackInUI, completion_callback);
   if (!blob_data_handle) {
-    base::PostTaskWithTraits(FROM_HERE, {BrowserThread::UI},
-                             base::BindOnce(callback, nullptr, 0));
+    atom::util::Promise::RejectPromise(std::move(promise),
```

Comment:
The code is already under `atom` namespace, so the `atom::` prefix is not needed.